play-client/src/main/scala/endpoints/EndpointPlayClient.scala

```diff
@@ -28,21 +28,20 @@ class EndpointPlayClient(wsClient: WSClient)(implicit ec: ExecutionContext) exte
   /**
     * A function that, given an `A` information and a `WSRequest`, eventually returns a `WSResponse`
     */
-  type RequestEntity[A] = (A, WSRequest) => Future[WSResponse]
+  type RequestEntity[A] = (A, WSRequest) => WSRequest
+
+  override val emptyRequestEntity: RequestEntity[Unit] = { case (_, req) => req }
 
-  def get[A, B](url: Url[A], headers: RequestHeaders[B])(implicit tupler: Tupler[A, B]): Request[tupler.Out] =
-    (ab: tupler.Out) => {
-      val (a, b) = tupler.unapply(ab)
-      val wsRequest = wsClient.url(url.encode(a))
-      headers(b, wsRequest).get()
-    }
 
-  def post[A, B, C, AB](url: Url[A], entity: RequestEntity[B], headers: RequestHeaders[C])(implicit tuplerAB: Tupler.Aux[A, B, AB], tuplerABC: Tupler[AB, C]): Request[tuplerABC.Out] =
+  override def request[A, B, C, AB](
```

Comment:
> IMHO, this is not really a problem because in such a case at some point the compiler will complain that some method is not implemented.

In normal code when you're using and implementing a thing at the same moment it's ok. But when you can implement something in module X and end up with error few days later in module Y. 

> On the other hand, if you always put the override keyword you can unintentionally override a method that was already implemented.

Just curious, could you tell more about such case? If I want to implement some abstract method how can I miss the fact that it is already implemented somewhere? Do you mean some complicated inheritance hierarchy? 

Other thing is that IMHO presence of `override` increase readability, so you know that signature is defined somewhere else and you cannot just change it. 

I think both position have some advantages and disadvantages, and it will be hard to find consensus. If it's important to you just tell me and I will gladly change it without further discussion :)